plugin/registry.go

```diff
@@ -36,36 +38,40 @@ var pluginNameRegex = regexp.MustCompile("^[0-9a-zA-Z_-]+$")
 // RegisterPlugin initializes the given plugin and adds it to the registry if
 // initialization was successful.
 func (r *Registry) RegisterPlugin(root Root, config map[string]interface{}) error {
-	registerPlugin := func() {
+	registerPlugin := func(checkName bool) {
+		r.mux.Lock()
+		if checkName {
```

Comment:
Think `checkName` => `initSucceeded` -- latter provides more context about what's going on here. It took me a bit to understand why you did this.